nxdrive/direct_edit.py

```diff
@@ -877,15 +890,19 @@ def _handle_upload_queue(self) -> None:
                 else:
                     # Try again in 30s
                     log.exception(f"Direct Edit unhandled HTTP error for ref {ref!r}")
-                    self._handle_upload_error(ref, os_path)
+                    self._handle_upload_error(ref, os_path, remote)
             except Exception:
                 # Try again in 30s
                 log.exception(f"Direct Edit unhandled error for ref {ref!r}")
-                self._handle_upload_error(ref, os_path)
+                self._handle_upload_error(ref, os_path, remote)
 
-    def _handle_upload_error(self, ref: Path, os_path: Path, /) -> None:
+    def _handle_upload_error(self, ref: Path, os_path: Path, remote: Remote, /) -> None:
         """Retry the upload if the number of attempts is below *._error_threshold* else discard it."""
         self._upload_errors[ref] += 1
+        if "directEdit.error.count" not in self._file_metrics[ref]:
+            self._file_metrics[ref]["directEdit.error.count"] = 1
```

Comment:
The whole thing could be simplified using a `defaultdict` when setting the empty dict:
```python
self._file_metrics[ref] = defaultdict(int)
```
Then you can replace that code with one call to
```python
self._file_metrics[ref]["directEdit.error.count"] += 1
```